app/src/main/java/com/alphawallet/app/ui/WalletsActivity.java

```diff
@@ -150,7 +168,7 @@ private void initViews()
         list = findViewById(R.id.list);
         list.setLayoutManager(new LinearLayoutManager(this));
 
-        adapter = new WalletsSummaryAdapter(this, this::onSetWalletDefault, viewModel.getWalletInteract(), preferenceRepository.isActiveMainnet());
+        adapter = new WalletsSummaryAdapter(this, this::onSetWalletDefault, viewModel.getWalletInteract(), preferenceRepository.isActiveMainnet(), viewModel.tokensService, viewModel.tokenRepository);
         list.setAdapter(adapter);
```

Comment:
No need to pass tokensService or TokenRepository any more. Make these values private again - also, do not get these services directly, use a getter like ```getTokensService() { return tokensService; }```